src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/Node.cs

```diff
@@ -55,6 +57,186 @@ public int GroupIndex
             get; set;
         }
 
+        /// <summary>
+        /// Gets Node display name
+        /// </summary>
+        public virtual string DisplayName
+        {
+            get
+            {
+                if (this.Operation == Operation.Unknown)
+                {
+                    return String.Empty;
+                }
+
+                // The display name can consist of two lines
+                // The first line is the Physical name and the physical kind in parenthesis
+                // The second line should contains either Object value or LogicalOp name.
+                // The second line should not show the same content as the first line.
+
+                string firstLine = this["PhysicalOp"] as string;
+                if (firstLine == null)
+                {
+                    if (this.Operation == null)
+                    {
+                        return String.Empty;
+                    }
+
+                    firstLine = this.Operation.DisplayName;
+                }
+
+                // Check if the PhysicalOp is specialized to a specific kind
+                string firstLineAppend = this["PhysicalOperationKind"] as string;
+                if (firstLineAppend != null)
+                {
+                    firstLine = String.Format(CultureInfo.CurrentCulture, "{0} {1}", firstLine, SR.Parenthesis(firstLineAppend));
+                }
+
+
+                string secondLine;
+
+                object objectValue = this["Object"];
+                if (objectValue != null)
+                {
+                    secondLine = GetObjectNameForDisplay(objectValue);
+                }
+                else
+                {
+                    secondLine = this["LogicalOp"] as string;
+                    if (secondLine != null)
+                    {
+                        if (secondLine != firstLine)
+                        {
+                            // Enclose logical name in parenthesis.
+                            secondLine = SR.Parenthesis(secondLine);
+                        }
+                        else
+                        {
+                            // Don't show the second line if its value is the same as on the first line.
+                            secondLine = null;
+                        }
+                    }
+                }
+
+                return secondLine == null || secondLine.Length == 0
+                    ? firstLine
+                    : String.Format(CultureInfo.CurrentCulture, "{0}\n{1}", firstLine, secondLine);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets node name (visible through AA)
+        /// </summary>
+        public string Name2
+        {
+            get
+            {
+                // Make AA names compatible with the previos implementation
```

Comment:
What does this mean? What previous implementation? Remember people reading this code have no context for where this came from originally so explaining why this stuff is actually necessary is better.

Reply:
I got these AA comments from SSMS. Will remove this prop as it is not used currently.